ext/js/lib/js/require_remote/response_handler.rb

```diff
@@ -0,0 +1,48 @@
+module JS
+  class RequireRemote
+    # Execute the body of the response and record the URL.
+    class ResponseHandler
+      # The around_eval parameter is a block that is executed before and after the body of the response.
+      # The block is passed the URL of the response.
+      def initialize(around_eval)
+        @around_eval = around_eval
+      end
+
+      # Execute the body of the response and record the URL.
+      def execute_and_record(response, location)
+        if response[:status].to_i == 200
+          # Check if the redirected URL has already evaluated.
+          return false if url_evaluated?(response[:url].to_s)
+
+          code = response.text().await.to_s
+          evaluate(code, location)
```

Comment:
TODO: The URL to be used as the basis for the relative path is the URL after the redirect.